### PR TITLE
Add aws-iam-authenticator recipe

### PR DESCRIPTION
**Pull request by edisongustavo**
Adding aws-iam-authenticator to conda-forge.

Currently, the recommended way to install, as per the AWS docs, is to download the executable directly from S3: https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html

So I thought in at least providing the binaries through conda-forge.

### CODE REVIEW

File: recipes/aws-iam-authenticator/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (edisongustavo) [inline recipes/aws-iam-authenticator/LICENSE.txt]: I've copied the License file from the original repo: https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/LICENSE

Is this the recommended way?

Comment (marcelotrevisani) [reply recipes/aws-iam-authenticator/LICENSE.txt]: If the license is already being packaged with the software, it is not necessary

File: recipes/aws-iam-authenticator/download-and-calculate-sha256.bash [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+version="1.11.5"
```
Comment (marcelotrevisani) [inline recipes/aws-iam-authenticator/download-and-calculate-sha256.bash]: I think you don't need this file right!?

File: recipes/aws-iam-authenticator/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  url: https://amazon-eks.s3-us-west-2.amazonaws.com/{{ version }}/{{ version_date }}/bin/windows/amd64/aws-iam-authenticator.exe  # [win]
+  sha256: d58439c548433adcda98e695be53e526ba940a4b9c44fb9a05d92cd495cdd47f  # [win]
+
+build:
+  number: 0
+
+test:
+  commands:
+    - aws-iam-authenticator -h  # [linux or osx]
```
Comment (marcelotrevisani) [inline recipes/aws-iam-authenticator/meta.yaml]: You can change it to just 
```
# [unix]
```

File: recipes/aws-iam-authenticator/build.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+chmod a+x aws-iam-authenticator
+mv aws-iam-authenticator $PREFIX/bin
```
Comment (marcelotrevisani) [inline recipes/aws-iam-authenticator/build.sh]: Is this package just one script? Or are you moving a binary directly?

File: recipes/aws-iam-authenticator/LICENSE.txt (hunk 1 above)
Comment (edisongustavo) [reply recipes/aws-iam-authenticator/LICENSE.txt]: I can see that the linter complained. Should I remove the LICENSE file and the `license_file` from the `meta.yaml` instead?

File: recipes/aws-iam-authenticator/download-and-calculate-sha256.bash (hunk 2 above)
Comment (edisongustavo) [reply recipes/aws-iam-authenticator/download-and-calculate-sha256.bash]: No, this is mostly an utility.

File: recipes/aws-iam-authenticator/build.sh (hunk 4 above)
Comment (edisongustavo) [reply recipes/aws-iam-authenticator/build.sh]: I'm moving a binary directly.

I've tried to simply use the `script` clause in the `meta.yaml` but I was not successful.

File: recipes/aws-iam-authenticator/download-and-calculate-sha256.bash (hunk 2 above)
Comment (edisongustavo) [reply recipes/aws-iam-authenticator/download-and-calculate-sha256.bash]: Do you think I should just remove it?

Comment (marcelotrevisani) [reply recipes/aws-iam-authenticator/download-and-calculate-sha256.bash]: I think would be better if you remove it, if you don't mind

File: recipes/aws-iam-authenticator/LICENSE.txt (hunk 1 above)
Comment (edisongustavo) [reply recipes/aws-iam-authenticator/LICENSE.txt]: > If the license is already being packaged with the software

It's not, so I will keep it then.

File: recipes/aws-iam-authenticator/build.sh (hunk 4 above)
Comment (marcelotrevisani) [reply recipes/aws-iam-authenticator/build.sh]: Did you try:
```
script:
  - chmod a+x aws-iam-authenticator  # [unix]
  - mv aws-iam-authenticator $PREFIX/bin  # [unix]
  - xcopy aws-iam-authenticator.exe %LIBRARY_BIN%  # [win]
```

I don't mind to have two files to execute it. It is not necessary to change

File: recipes/aws-iam-authenticator/LICENSE.txt (hunk 1 above)
Comment (marcelotrevisani) [reply recipes/aws-iam-authenticator/LICENSE.txt]: Yes, just remove it and should be fine

File: recipes/aws-iam-authenticator/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  number: 0
+
+test:
+  commands:
+    - aws-iam-authenticator -h  # [unix]
+    - aws-iam-authenticator.exe -h  # [win]
+
+about:
+  home: https://github.com/kubernetes-sigs/aws-iam-authenticator
+  license: Apache License 2.0
```
Comment (marcelotrevisani) [inline recipes/aws-iam-authenticator/meta.yaml]: The linter is complain because of the word ``License`` here
It should be ``Apache 2.0``

Comment (marcelotrevisani) [reply recipes/aws-iam-authenticator/meta.yaml]: I mean
```
license: Apache 2.0
```

File: recipes/aws-iam-authenticator/build.sh (hunk 4 above)
Comment (edisongustavo) [reply recipes/aws-iam-authenticator/build.sh]: That would probably work.

What would you prefer?

File: recipes/aws-iam-authenticator/meta.yaml (hunk 5 above)
Comment (edisongustavo) [reply recipes/aws-iam-authenticator/meta.yaml]: Yes, that's it ;)

File: recipes/aws-iam-authenticator/build.sh (hunk 4 above)
Comment (marcelotrevisani) [reply recipes/aws-iam-authenticator/build.sh]: You can let this as it is, using two separated files. That is fine :)

**Comment by edisongustavo** [comment]
@marcelotrevisani all checks passed (except for Azure, which I was told in Gitter it is safe to ignore).

Would you mind merging this?

**Comment by marcelotrevisani** [comment]
Sure! Thanks for adding this to conda-forge ! :)

**Pull request merged**